src/watchmaker/__init__.py

```diff
@@ -283,11 +283,23 @@ def __init__(self, arguments):
             (k.lstrip('-').replace('-', '_'), v) for k, v in zip(
                 *[iter(extra_arguments)] * 2)
         ))
-        # Set self.worker_args, removing `None` values from worker_args
-        self.worker_args = dict(
-            (k, yaml.safe_load(v)) for k, v in worker_args.items()
-            if v != Arguments.DEFAULT_VALUE
-        )
+
+        try:
+            # Set self.worker_args, removing `None` values from worker_args
+            self.worker_args = dict(
+                (k, yaml.safe_load(v)) for k, v in worker_args.items()
+                if v != Arguments.DEFAULT_VALUE
+            )
+        except yaml.YAMLError as exc:
+            if hasattr(exc, "problem_mark"):
+                msg = (
+                    "Failed to parse argument value as YAML. Check the format "
+                    "and/or properly quote the value when using the CLI to "
+                    "account for shell interpolation. YAML error: {0}"
+                ).format(str(exc))
```

Comment:
Just a minor formatting comment.  Can you add newlines so the error message prints similar to how it's in the code?


Something like this?
```
raise InvalidValue(msg) watchmaker.exceptions.InvalidValue:
Failed to parse argument value as YAML. Check the format
and/or properly quote the value when using the CLI to
account for shell interpolation.
YAML error: found undefined alias 'RHEL'
in "<unicode string>", line 1, column 1:
    *RHEL
    ^
```

Reply:
Eh, I'd rather not? The line breaks in the code are dictated by the indent of the code and the lint check for 80 chars. Trying to force those same line breaks on the shell, which is itself dependent on the terminal's line-wrapping logic, tends to result in unexpected pain... Improving how watchmaker manages its output for user consumption is a larger project. It does nothing now, just uses "raise" to bail. This kind of output handling would better be managed in a set of exception classes, instead of trying to format individual exception messages...

Reply:
Okay.  Yeah, handling is the outputs as a whole is better.  Good for me.